### PR TITLE
Feature/adding viewmodel

### PR DESCRIPTION
**Pull request by UmairShaik7**
MVVM_structure

### CODE REVIEW

**Comment by UmairShaik7** [comment]
please review this brach

File: app/src/main/java/com/example/weatherwatchapp/AppModule.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.example.weatherwatchapp
+
+import com.example.weatherwatchapp.ui.main.MainActivityViewModel
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.dsl.module
+
+private const val BASE_URL: String = "Any"
```
Comment (asdheeraj) [inline app/src/main/java/com/example/weatherwatchapp/AppModule.kt]: what is the use of this variable?

File: app/src/main/java/com/example/weatherwatchapp/RepoModule.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+class APIKeyInterceptor : Interceptor {
+
+    @Throws(IOException::class)
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val original: Request = chain.request()
+        val originalHttpUrl: HttpUrl = original.url
+
+        val url = originalHttpUrl.newBuilder()
+            .addQueryParameter("appid", "4b0151007b2b506fe4573e9067c96926")
```
Comment (asdheeraj) [inline app/src/main/java/com/example/weatherwatchapp/RepoModule.kt:55]: Are these required for fetching the weather?

File: app/src/main/java/com/example/weatherwatchapp/repo/MainRepository.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.example.weatherwatchapp.repo
+
+import com.example.weatherwatchapp.helper.NetworkResult
+import com.example.weatherwatchapp.helper.NetworkResult.Success
+import com.example.weatherwatchapp.repo.data.WeatherAPIData
+
+class MainRepository(private var remoteSource: RemoteNetworkSource) {
```
Comment (asdheeraj) [inline app/src/main/java/com/example/weatherwatchapp/repo/MainRepository.kt]: Please rename this to WeatherRepository

File: app/src/main/java/com/example/weatherwatchapp/repo/data/WeatherAPIData.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    var main: String? = null,
+
+    @SerializedName("description")
+    @Expose
+    var description: String? = null,
+
+    @SerializedName("icon")
+    @Expose
+    var icon: String? = null
+) : Parcelable
```
Comment (asdheeraj) [inline app/src/main/java/com/example/weatherwatchapp/repo/data/WeatherAPIData.kt:44]: Any reason why we are making this extend parcelable?

**Review by asdheeraj — COMMENTED** [review]
Just check if we can rename the MainRepo/Acrtivity to Weather Activity/WeatherRepo

File: app/src/main/java/com/example/weatherwatchapp/RepoModule.kt (hunk 2 above)
Comment (UmairShaik7) [reply app/src/main/java/com/example/weatherwatchapp/RepoModule.kt:55]: API keys for weather api. Mandatory

File: app/src/main/java/com/example/weatherwatchapp/AppModule.kt (hunk 1 above)
Comment (UmairShaik7) [reply app/src/main/java/com/example/weatherwatchapp/AppModule.kt]: we don't need this variable removed it

File: app/src/main/java/com/example/weatherwatchapp/repo/data/WeatherAPIData.kt (hunk 4 above)
Comment (UmairShaik7) [reply app/src/main/java/com/example/weatherwatchapp/repo/data/WeatherAPIData.kt:44]: we need this object. we will pass as intent to watch

**Pull request merged**